File: src/com/eagle/programmar/JavaP/Blocks/JavaP_CodeExceptions.java
// Copyright Eagle Legacy Modernization, LLC, 2010-date
// Original author: Steven A. O'Hara, Nov 4, 2015

package com.eagle.programmar.JavaP.Blocks;

import com.eagle.programmar.JavaP.Terminals.JavaP_EndOfLine;
import com.eagle.programmar.JavaP.Terminals.JavaP_Keyword;
import com.eagle.programmar.JavaP.Terminals.JavaP_Number;
import com.eagle.programmar.JavaP.Terminals.JavaP_QualifiedName;
import com.eagle.tokens.TokenList;
import com.eagle.tokens.TokenSequence;
import com.eagle.tokens.punctuation.PunctuationColon;

public class JavaP_CodeExceptions extends TokenSequence
{
	public @S(10) JavaP_Keyword EXCEPTION = new JavaP_Keyword("Exception");
	public @S(20) JavaP_Keyword TABLE = new JavaP_Keyword("table");
	public @S(30) PunctuationColon colon;
	public @S(40) JavaP_EndOfLine eoln1;
	
	public @S(50) JavaP_Keyword FROM = new JavaP_Keyword("from");
	public @S(60) JavaP_Keyword TO = new JavaP_Keyword("to");
	public @S(70) JavaP_Keyword TARGET = new JavaP_Keyword("target");
	public @S(80) JavaP_Keyword TYPE = new JavaP_Keyword("type");
	public @S(90) JavaP_EndOfLine eoln2;

	public @S(100) @OPT TokenList<JavaP_CodeExceptionEntry> entries;
	
	public static class JavaP_CodeExceptionEntry extends TokenSequence
	{
		public @S(10) JavaP_Number from;
		public @S(20) JavaP_Number to;
		public @S(30) JavaP_Number target;
		public @S(40) @OPT JavaP_Keyword CLASS = new JavaP_Keyword("Class");
		public @S(50) JavaP_QualifiedName name;
		public @S(60) JavaP_EndOfLine eoln;
	}
}

File: csrc/com/eagle/programmar/JavaP/Blocks/JavaP_CodeExceptions.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// Purchase a Premium Edition license at:
// https://www.tangiblesoftwaresolutions.com/order/order-java-to-csharp.html
// ====================================================================================================

// Copyright Eagle Legacy Modernization, LLC, 2010-date
// Original author: Steven A. O'Hara, Nov 4, 2015

namespace com.eagle.programmar.JavaP.Blocks
{
	using JavaP_EndOfLine = com.eagle.programmar.JavaP.Terminals.JavaP_EndOfLine;
	using JavaP_Keyword = com.eagle.programmar.JavaP.Terminals.JavaP_Keyword;
	using JavaP_Number = com.eagle.programmar.JavaP.Terminals.JavaP_Number;
	using JavaP_QualifiedName = com.eagle.programmar.JavaP.Terminals.JavaP_QualifiedName;
	using TokenList = com.eagle.tokens.TokenList;
	using TokenSequence = com.eagle.tokens.TokenSequence;
	using PunctuationColon = com.eagle.tokens.punctuation.PunctuationColon;

	public class JavaP_CodeExceptions : TokenSequence
	{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.JavaP.Terminals.JavaP_Keyword EXCEPTION = new com.eagle.programmar.JavaP.Terminals.JavaP_Keyword("Exception");
		public JavaP_Keyword EXCEPTION = new JavaP_Keyword("Exception");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) com.eagle.programmar.JavaP.Terminals.JavaP_Keyword TABLE = new com.eagle.programmar.JavaP.Terminals.JavaP_Keyword("table");
		public JavaP_Keyword TABLE = new JavaP_Keyword("table");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) com.eagle.tokens.punctuation.PunctuationColon colon;
		public PunctuationColon colon;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(40) com.eagle.programmar.JavaP.Terminals.JavaP_EndOfLine eoln1;
		public JavaP_EndOfLine eoln1;

// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(50) com.eagle.programmar.JavaP.Terminals.JavaP_Keyword FROM = new com.eagle.programmar.JavaP.Terminals.JavaP_Keyword("from");
		public JavaP_Keyword FROM = new JavaP_Keyword("from");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(60) com.eagle.programmar.JavaP.Terminals.JavaP_Keyword TO = new com.eagle.programmar.JavaP.Terminals.JavaP_Keyword("to");
		public JavaP_Keyword TO = new JavaP_Keyword("to");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(70) com.eagle.programmar.JavaP.Terminals.JavaP_Keyword TARGET = new com.eagle.programmar.JavaP.Terminals.JavaP_Keyword("target");
		public JavaP_Keyword TARGET = new JavaP_Keyword("target");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(80) com.eagle.programmar.JavaP.Terminals.JavaP_Keyword TYPE = new com.eagle.programmar.JavaP.Terminals.JavaP_Keyword("type");
		public JavaP_Keyword TYPE = new JavaP_Keyword("type");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(90) com.eagle.programmar.JavaP.Terminals.JavaP_EndOfLine eoln2;
		public JavaP_EndOfLine eoln2;

// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(100) @OPT TokenList<JavaP_CodeExceptionEntry> entries;
		public  OPT;

		public class JavaP_CodeExceptionEntry : TokenSequence
		{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.JavaP.Terminals.JavaP_Number from;
			public JavaP_Number from;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) com.eagle.programmar.JavaP.Terminals.JavaP_Number to;
			public JavaP_Number to;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) com.eagle.programmar.JavaP.Terminals.JavaP_Number target;
			public JavaP_Number target;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(40) @OPT JavaP_Keyword CLASS = new com.eagle.programmar.JavaP.Terminals.JavaP_Keyword("Class");
			public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(50) com.eagle.programmar.JavaP.Terminals.JavaP_QualifiedName name;
			public JavaP_QualifiedName name;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(60) com.eagle.programmar.JavaP.Terminals.JavaP_EndOfLine eoln;
			public JavaP_EndOfLine eoln;
		}
	}

}
